bench-tps/src/bench_tps_client/tpu_client.rs

```diff
@@ -1,13 +1,21 @@
 use {
     crate::bench_tps_client::{BenchTpsClient, BenchTpsError, Result},
     solana_client::tpu_client::TpuClient,
+    solana_connection_cache::connection_cache::{
+        ConnectionManager, ConnectionPool, NewConnectionConfig,
+    },
     solana_sdk::{
         account::Account, commitment_config::CommitmentConfig, epoch_info::EpochInfo, hash::Hash,
         message::Message, pubkey::Pubkey, signature::Signature, transaction::Transaction,
     },
 };
 
-impl BenchTpsClient for TpuClient {
+impl<R, S, T> BenchTpsClient for TpuClient<R, S, T>
+where
+    R: ConnectionPool<NewConnectionConfig = T>,
```

Comment:
The variable names are already descriptive; e.g. `fn new(connection_manager: S, ...)`
There are also descriptive trait constraints, e.g. `R: ConnectionPool<NewConnectionConfig = T>,`
and I have further annotated each with comments where needed.

Reply:
If you could use a generic that isn't `T` for the config, it would cut some of the diff out of this PR, which would be a net positive, imo.

Reply:
> If you could use a generic that isn't T for the config, it would cut some of the diff out of this PR

There are just like ~10 lines out of 500+ lines diff that were (trivially) changed due that.

>  Since R, S, and T don't seem to have any relevance to the traits they impl, why not use something else?

I could have started with different letters if I knew, but given [the existing context](https://github.com/solana-labs/solana/pull/30128#discussion_r1098735584) (descriptive variable names, descriptive trait constraints and comments), I don't think changing one single letter vs another single letter is adding much or is a productive use of time here.

Reply:
Frankly, anything that reduces the diff of a big PR like this, and doesn't make them spend cycles figuring out why something unrelated has changed, makes it easier on reviewers. And I think anything that makes generics more intuitive makes the code less likely to be broken in the future.
So here you go: https://github.com/behzadnouri/solana/pull/116 Happy to push directly instead.

Reply:
merged, thank you